field/VerticalStaggerLoc.F90

```diff
@@ -0,0 +1,95 @@
+module mapl3g_VerticalStaggerLoc
+   implicit none
+   private
+
+   public :: VerticalStaggerLoc
+   public :: VERTICAL_STAGGER_NONE
+   public :: VERTICAL_STAGGER_EDGE
+   public :: VERTICAL_STAGGER_CENTER
+   public :: VERTICAL_STAGGER_INVALID
+
+   public :: operator(==)
+   public :: operator(/=)
+
+   public :: make_VerticalStaggerLoc
+
+   type :: VerticalStaggerLoc
+      private
+      integer :: id
+   contains
+      ! TODO: Convert to DTIO once compilers support allocatable internal files
+      procedure :: to_string
+   end type VerticalStaggerLoc
+
+   interface operator(==)
+      procedure are_equal
+   end interface operator(==)
+
+   interface operator(/=)
+      procedure are_not_equal
+   end interface operator(/=)
+
+   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_NONE = VerticalStaggerLoc(1)
+   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_EDGE = VerticalStaggerLoc(2)
+   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_CENTER = VerticalStaggerLoc(3)
+   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_INVALID = VerticalStaggerLoc(4)
```

Comment:
If you move the name string into the type, the instantiation would look like this:
```suggestion
   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_NONE = VerticalStaggerLoc(1, "VERTICAL_STAGGER_NONE")
   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_EDGE = VerticalStaggerLoc(2, "VERTICAL_STAGGER_EDGE")
   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_CENTER = VerticalStaggerLoc(3, "VERTICAL_STAGGER_CENTER")
   type(VerticalStaggerLoc), parameter :: VERTICAL_STAGGER_INVALID = VerticalStaggerLoc(4, "VERTICAL_STAGGER_INVALID")
```

Reply:
It would be nice, but ...